src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,246 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import measure_theory.measure.measure_space
+
+/-!
+# Weak convergence of (finite) measures
+
+This file will define the topology of weak convergence of finite measures and probability measures
+on topological spaces. The topology of weak convergence is the coarsest topology w.r.t. which
+for every bounded continuous `ℝ≥0`-valued function `f` on, the integration of `f` against the
+measure is continuous.
+
+TODOs:
+* Define the topologies (the current version only defines the types) via
+  `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`.
+* Prove that an equivalent definition of the topologies is obtained using bounded continuous
+  `ℝ`-valued functions instead.
+* Include the portmanteau theorem on characterizations of weak convergence of (Borel) probability
+  measures.
+
+## Main definitions
+
+The main definitions are the types `finite_measure α` and `probability_measure α`.
+
+TODO:
+* Define the topologies on the above types.
+
+## Main results
+
+None yet.
+
+TODO:
+* Portmanteau theorem.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+The topology of weak convergence of finite Borel measures will be defined using a mapping from
+`finite_measure α` to `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`, inheriting the topology from there.
+
+## References
+
+* [Billingsley, *Convergence of probability measures*][billingsley1999]
+
+## Tags
+
+weak convergence of measures, finite measure, probability measure
+
+-/
+
+noncomputable theory
+open measure_theory
+open set
+open filter
+open_locale topological_space ennreal nnreal
+
+namespace measure_theory
+
+section finite_measure
+/-!
+### Finite measures
+In this section, we define the `Type` of finite measures on a given measurable space. This type
+will be equipped with the topology of weak convergence of measures when the measurable space is
+a topological space equipped with its Borel sigma-algebra.
+-/
+
+variables {α : Type*} [measurable_space α]
+
+/-- Finite measures are defined as the subtype of measures that have the property of being finite
+measures (i.e., their total mass is finite). -/
+def finite_measure (α : Type*) [measurable_space α] : Type* :=
+{ μ : measure α // is_finite_measure μ }
+
+namespace finite_measure
+
+instance has_zero : has_zero (finite_measure α) :=
+{ zero := ⟨0, measure_theory.is_finite_measure_zero⟩ }
```

Comment:
```suggestion
{ zero := ⟨0, is_finite_measure_zero⟩ }
```
Now that you are working in the `measure_theory` namespace, you shouldn't need the prefix. Same thing for all other occurences of `measure_theory.foo`